src/main.rs

```diff
@@ -85,8 +84,9 @@ fn run() -> Result<()> {
         }
     };
 
-    if data.len() > MAX_DATA_SIZE {
-        bail!("the data size is too large, we limit the maximum size to 30 MiB to ensure TUI performance, you should try to reduce the read size");
+    let max_data_size = args.max_data_size.unwrap_or(cfg.data.max_data_size) * 1024 * 1024;
+    if data.len() > max_data_size {
+        bail!("the data size is too large, we limit the maximum size to {} to ensure TUI performance, you should try to reduce the read size", humansize::format_size(max_data_size, humansize::BINARY));
```

Comment:
So we don't suggest them to use a larger value?

I'm fine with this, but I'm surprised the error message doesn't inform the user the configuration can be updated?

Reply:
Maybe change to:

```suggestion
        bail!("the data size is too large, we limit the maximum size to {} to ensure TUI performance, you should try to reduce the read size or use `--max-data-size` to change the limitation", humansize::format_size(max_data_size, humansize::BINARY));
```

Reply:
Yes something like this, but it also be via configuration, so it might need to be adapted

Reply:
```suggestion
        bail!("the data size is too large, we limit the maximum size to {} to ensure TUI performance, you should try to reduce the read size. HINT: You can use command line arg `--max-data-size` or config option `data.max_data_size` to modify this limitation", humansize::format_size(max_data_size, humansize::BINARY));
```

Reply:
LGTM 👍